romancal/datamodels/tests/test_datamodels.py

```diff
@@ -504,3 +510,81 @@ def test_make_file_with_index(filename, idx, expected_filename_result, tmp_path)
     result = make_file_with_index(file_path=filepath, idx=idx)
 
     assert result == str(tmp_path / expected_filename_result)
+
+
+def test_modelcontainer_works_properly_with_context_manager(tmp_path):
+    """Test that ModelContainer works correctly with context manager."""
+
+    filename1 = tmp_path / "img_1.asdf"
+    filename2 = tmp_path / "img_2.asdf"
+    # create ImageModel
+    img1 = utils.mk_level2_image()
+    img2 = utils.mk_level2_image()
+    img1.meta.filename = filename1
+    img2.meta.filename = filename2
+    # create datamodels from ImageModel
+    dm1 = rdm.ImageModel(img1)
+    dm2 = rdm.ImageModel(img2)
+    # save datamodels
+    dm1.save(filename1)
+    dm2.save(filename2)
+
+    # create ASN file that points to the ASDF files
+    asn_filepath = create_asn_file(tmp_path)
+
+    result_filenames = []
+
+    with ModelContainer(asn_filepath) as asn1:
+        result_filenames.extend(
+            tmp_path / datamodel.meta.filename for datamodel in asn1
+        )
+
+    expected_filenames = [filename1, filename2]
+    expected_filenames.sort()
+    result_filenames.sort()
+
+    assert all(e == r for e, r in zip(expected_filenames, result_filenames))
+    assert type(asn1) is ModelContainer
```

Comment:
Can the above code basically just be
with ModelContainer(asn_filepath, return_open=False, save_open=False ) as asn1:
    assert type(asn1) is ModelContainer

What does the rest of the code above do wrt the context managment?